connector/tests/src/test/java/oaebudt/dataspace/connector/ManagementApiTransferTest.java

```diff
@@ -0,0 +1,209 @@
+        final String assetId = createProviderAsset(dataAddressProperties);
+
+        final String transferProcessId = CONSUMER.requestAssetFrom(assetId, PROVIDER)
+                .withTransferType("HttpData-PULL")
+                .withCallbacks(Json.createArrayBuilder()
+                        .add(createCallback("http://localhost:%s/edr".formatted(consumerEdrReceiver.getPort()), true, Set.of("transfer.process.started")))
+                        .build())
+                .execute();
+
+        CONSUMER.awaitTransferToBeInState(transferProcessId, STARTED);
+
+        final var edrRequests = await().until(() -> consumerEdrReceiver.retrieveRecordedRequests(request("/edr")), it -> it.length > 0);
+
+        final var edr = new ObjectMapper().readTree(edrRequests[0].getBodyAsRawBytes()).get("payload").get("dataAddress").get("properties");
+
+        final var endpoint = edr.get(EDC_NAMESPACE + "endpoint").asText();
+        final var authCode = edr.get(EDC_NAMESPACE + "authorization").asText();
+
+        final var body = given()
+                .header("Authorization", authCode)
+                .when()
+                .get(endpoint)
+                .then()
+                .log().ifValidationFails()
+                .statusCode(200)
+                .extract().body().asString();
+
+        Assertions.assertThat(body).isEqualTo("data");
+
+        consumerEdrReceiver.stop();
+
+    }
+
+    private String createProviderAsset(final Map<String, Object> dataAddressProperties) {
+        final var assetId = UUID.randomUUID().toString();
+
+        final var noConstraintPolicyId = PROVIDER.createPolicyDefinition(PolicyFixtures.noConstraintPolicy());
+        PROVIDER.createAsset(assetId, Map.of("name", "description"), dataAddressProperties);
+        PROVIDER.createContractDefinition(assetId, UUID.randomUUID().toString(), noConstraintPolicyId, noConstraintPolicyId);
+        return assetId;
+    }
+
+    private JsonObject httpDataAddress(final String baseUrl) {
+        return Json.createObjectBuilder()
+                .add(TYPE, EDC_NAMESPACE + "DataAddress")
+                .add(EDC_NAMESPACE + "type", "HttpData")
+                .add(EDC_NAMESPACE + "baseUrl", baseUrl)
+                .build();
+    }
+
+    public JsonObject createCallback(final String url, final boolean transactional, final Set<String> events) {
+        return Json.createObjectBuilder()
+                .add(TYPE,  "CallbackAddress")
+                .add("transactional", transactional)
+                .add("uri", url)
+                .add("events", events
+                        .stream()
+                        .collect(Json::createArrayBuilder, JsonArrayBuilder::add, JsonArrayBuilder::add)
+                        .build())
+                .build();
+    }
+
+
+    private static int getVaultPort() {
+
+        if (!vaultContainer.isRunning()) {
+            vaultContainer.start();
+        }
+
+        return vaultContainer.getMappedPort(VAULT_PORT);
```

Comment:
This could be avoided if the vault (but, also postgres) are wrapped behind a junit extensions, and they are registered with an order, e.g. [as it is done in the e2e tests in the EDC project](https://github.com/eclipse-edc/Connector/blob/c73d1477b7c4a327bb6fb4c79f945f15e1e8237a/system-tests/e2e-transfer-test/runner/src/test/java/org/eclipse/edc/test/e2e/TransferPullEndToEndTest.java#L498)
At that point we can be confident that the vault/postgres are already started when the connector is starting, the the `getFirstMappedPort()` (that will be wrapped by the extension class as well) is not giving errors.